### ISSUE TITLE
[Scarpet suggestion] Allow to access offline player's inventories

### ISSUE THREAD

**Issue by Firigion** [gnembon/fabric-carpet#808]
You currently can't access offline player's inventories nor enderchests, both of which would be useful for some moderation tools that could easily be written in scarpet. I don't see a real use outside this, though, so I'd understand if this gets ignored.

**Comment by DillonB07** [comment]
[https://github.com/gnembon/scarpet/](https://github.com/gnembon/scarpet/)

**Comment by DragonEggBedrockBreaking** [comment]
> Try posting here instead:
> https://github.com/gnembon/scarpet/issues

No. This is the carpet mod repo, for asking for new features to be added to scarpet. The scarpet repo is for scarpet apps.

**Comment by Firigion** [comment]
Hm? A deleted comment?

**Comment by DragonEggBedrockBreaking** [comment]
> Hm? A deleted comment?

Yes

**Comment by Ghoulboy78** [comment]
It would require digging into player data file, and returning a blank inventory if not. Problem is that if the inventory is going to be writable, you could end up generating a bunch of player data files, which may cause quite some IO lag, so imo it should be read-onnly, but then again I am aware that that makes it far less useful...

**Comment by Firigion** [comment]
Still useful, tho. You at leas know _where_ the stuff is, if you need to know. Issue is then, a read only-inventory would be a new thing, there's currently no ready-only invs I don't think.

**Comment by altrisi** [comment]
I guess if you don't need the access regularly you could just move the player's nbt file inside your app's data folder temporarily and parse their contents as NBT, but yeah, it would probably be nice to have a way to access them directly.
